src/bp/ui-admin/src/Pages/Workspace/Bots/RollbackBotModal.jsx

```diff
@@ -0,0 +1,95 @@
+import React, { Component } from 'react'
+import { Button, Modal, ModalHeader, ModalBody, FormGroup, Label } from 'reactstrap'
+import { MdReplay } from 'react-icons/lib/md'
+import Select from 'react-select'
+
+import api from '../../../api'
+
+const defaultState = {
+  revisions: [],
+  selectedRev: null
+}
+
+class RollbackBotModal extends Component {
+  state = { ...defaultState }
+
+  async componentDidUpdate(prevProps) {
+    if (this.props.botId && prevProps.botId !== this.props.botId) {
+      await this.fetchRevisions()
+    }
+  }
+
+  fetchRevisions = () => {
+    return api
+      .getSecured()
+      .get(`/admin/bots/${this.props.botId}/revisions`)
+      .then(({ data }) => {
+        this.setState({ revisions: data.payload.revisions })
+      })
+  }
+
+  selectRev = ({ value: selectedRev }) => {
+    this.setState({ selectedRev })
+  }
+
+  rollback = () => {
+    if (
+      window.confirm(
+        `Are you sure you want to rollback bot ${this.props.botId}? All its content and flows will be overwritten`
+      )
+    ) {
+      api
+        .getSecured()
+        .post(`/admin/bots/${this.props.botId}/rollback`, { revision: this.state.selectedRev })
+        .then(() => {
+          this.props.onRollbackSuccess && this.props.onRollbackSuccess()
+          this.props.toggle()
+          this.setState({ ...defaultState })
+        })
+    }
+  }
+
+  focus = () => {
+    this.select.focus()
+  }
+
+  render() {
+    return (
+      <Modal isOpen={this.props.isOpen} toggle={this.props.toggle} fade={false} onOpened={this.focus} autoFocus={false}>
+        <ModalHeader toggle={this.props.toggle}>Rollback bot {this.props.botId || ''}</ModalHeader>
+        <ModalBody>
+          <p>Select from the available revisions below</p>
+          <FormGroup>
+            <Label for="name">
+              <strong>Revisions</strong>
+            </Label>
+            <Select
+              tabIndex="1"
+              ref={el => (this.select = el)}
+              value={
+                this.state.selectedRev
+                  ? { label: this.state.selectedRev.replace('.tgz', ''), value: this.state.selectedRev }
+                  : null
+              }
+              options={this.state.revisions.map(rev => ({ label: rev.replace('.tgz', ''), value: rev }))}
```

Comment:
Rendering should have minimum logic. Everytime something changes on the page, everything here needs to be recomputed. I suggest putting your map() in the fetchRevisions method, and store in the state only the list of finished options